src/BaseDocumentSerializer.ts

```diff
@@ -1,253 +1,264 @@
-import schemas from 'part:@sanity/base/schema'
 import blocksToHtml, { h } from '@sanity/block-content-to-html'
 import { defaultStopTypes, customSerializers } from './BaseSerializationConfig'
 import { ObjectField, SanityDocument } from '@sanity/types'
-import { Serializer } from './types'
+import { Serializer, TranslationLevel } from './types'
+import Schema from '@sanity/schema'
 import clone from 'just-clone'
 
-/*
- * Helper function that allows us to get metadata (like `localize: false`) from schema fields.
- */
-const getSchema = (name: string) =>
-  schemas._original.types.find((s: ObjectField) => s.name === name)
-
-/*
- * Main parent function: finds fields to translate, and feeds them to appropriate child serialization
- * methods.
- */
-const serializeDocument = (
-  doc: SanityDocument,
-  translationLevel: string = 'document',
-  baseLang = 'en',
-  stopTypes = defaultStopTypes,
-  serializers = customSerializers
-) => {
-  let filteredObj: Record<string, any> = {}
-
-  if (translationLevel === 'field') {
-    filteredObj = languageObjectFieldFilter(doc, baseLang)
-  } else {
-    filteredObj = fieldFilter(doc, getSchema(doc._type).fields, stopTypes)
-  }
+type SerializerClosure = (scheas: Schema) => Serializer
+
+export const BaseDocumentSerializer: SerializerClosure = (schemas: Schema) => {
+  /*
+   * Helper function that allows us to get metadata (like `localize: false`) from schema fields.
+   */
+  // FIXME: Need to cast this to `any` as the defined type TypeDef does not
+  // match what the existing code is expecting on these objects
+  const getSchema = (name: string) =>
+    schemas._original.types.find(s => s.name === name) as any
```

Comment:
This felt a bit weird having to do, but could be a type error in the Schema._original field. It wanted to have it be `TypeDef` which did not match the code. I tried to coerce it into `SchemaType` but that did not work either. Therefor `any`